datatypes/src/dataset.rs

```diff
@@ -64,3 +61,297 @@ impl From<ExternalDataId> for DataId {
         DataId::External(value)
     }
 }
+
+pub const NAME_DELIMITER: char = ':';
+pub const SYSTEM_NAMESPACE: &str = "_";
+pub const SYSTEM_PROVIDER: &str = "_";
+
+/// The user-facing identifier for loadable data.
```

Comment:
shouldn't this be part of the services crate then?

Reply:
No, this should be part of the place where the API resides, since it translates `_` to None.
Since the API is at datatypes as well as the services crate, which is a bad design, we have to leave it there.